environment.go

```diff
@@ -467,6 +508,39 @@ func (e *envState) GetSender(key SenderKey) (send.Sender, error) {
 	return sender, nil
 }
 
+func (e *envState) RegisterCloser(name string, closer func(context.Context) error) {
+	e.mu.Lock()
+	defer e.mu.Unlock()
+
+	e.closers[name] = closer
+}
+
+func (e *envState) Close(ctx context.Context) error {
+	e.mu.RLock()
+	defer e.mu.RUnlock()
+
+	// TODO we could, in the future call all closers in but that
+	// would require more complex waiting and timeout logic
+
+	deadline, _ := ctx.Deadline()
+	catcher := grip.NewBasicCatcher()
+	for name, closer := range e.closers {
+		if closer == nil {
+			continue
+		}
+
+		grip.Info(message.Fields{
+			"message":      "calling closer",
+			"closer":       name,
+			"timeout_secs": time.Since(deadline),
```

Comment:
This confuses me a bit. This is time.Now().Sub(deadline), so won't the first run of this have a negative time? Did you mean time.Until(deadline)?

Additionally, don't you need to set a new value for deadline on each iteration?

Reply:
presumably if there's a timeout on this context the deadline will be in the future, so this will report the remaining time for each call.